src/cargo/util/to_url.rs

```diff
@@ -17,11 +18,28 @@ impl<'a> ToUrl for &'a Url {
     }
 }
 
+/// Uses `url::UrlParser` to parse a `str` into a `Url`.
+///
+/// The current working directory is used to support relative paths.
 impl<'a> ToUrl for &'a str {
     fn to_url(self) -> Result<Url, String> {
-        UrlParser::new().scheme_type_mapper(mapper).parse(self).map_err(|s| {
-            format!("invalid url `{}`: {}", self, s)
-        })
+        let base_url = determine_base_url();
```

Comment:
I'd appreciate suggestions on a better name for both the method and local variable.